tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -203,11 +203,13 @@ Status ExecuteThunks(const std::string& module_name,
       [&] { return absl::StrCat(module_name, ":XLA GPU module"); },
       tsl::profiler::TraceMeLevel::kInfo);
 
+  ScopedAnnotation annotation(module_name);
+
```

Comment:
IMO, this should be lifted outside of the loop (not only for the nvtx case), however without remove the the hlo module from tf-profiler path and add a layer here, this means the module name (which is quite long) will appear in the final annotation of each event twice for tf- profiler. I am not comfortable with this change.

Reply:
It is already out of the for loop.